File: Problems/053-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        l_nums = len(nums)
        ans = nums[0]
        for i in range(1, l_nums):
            nums[i] = nums[i] + max(nums[i - 1], 0)
            ans = max(nums[i], ans)
        return ans

File: Problems/053-maximum-subarray/maximum-subarray.go
package problem053

func maxSubArray(nums []int) int {
	ans := nums[0]

	for i := 1; i < len(nums); i++ {
		nums[i] = nums[i] + max(0, nums[i-1])
		ans = max(ans, nums[i])
	}

	return ans
}

func max(i, j int) int {
	if i > j {
		return i
	}
	return j
}
